descwl_coadd/interp.py

```diff
@@ -132,6 +133,78 @@ def interp_image_nocheck(image, bad_msk):
     return interp_image
 
 
+class CTInterpolator(object):
+    """
+    A class wrapping the interp_image_nocheck function to do
+    an inplace interp and sets the INTRP bit
+
+    This conforms to the interface required for the interpolator
+    sent to descwl_coadd.coadd.warp_exposures
+
+    This is a "functor" meaning the object can be called
+        interpolator(exposure)
+    """
+    def __init__(self, bad_mask_planes=FLAGS2INTERP):
+        self.bad_mask_planes = bad_mask_planes
+
+    def __call__(self, exp):
```

Comment:
To make this consistent with a `Task` interface that you might want to test swapping out, I'd recommend this as `run` instead of `__call__`.

Reply:
Are you saying that the input interpolator, if it was implemented by DM, would not be a functor but would instead use a `.run` method?

Reply:
Yes, that's correct. And that's because that is the standard interface for any `Task`.

1-D interpolator example (one that you don't want to use): https://github.com/lsst/pipe_tasks/blob/d598c5cd71ea7e32b599949b6d49c719ec281f19/python/lsst/pipe/tasks/interpImage.py#L131

CloughTocher interpolator (one you could use in your test, potentially):
https://github.com/lsst/meas_algorithms/blob/78056777b346eee9046fe87b60526f6edb5c8b0a/python/lsst/meas/algorithms/cloughTocher2DInterpolator.py#L69